bin/lint_tool_file_names.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+# Prohibit internal shell script file names snake_case
+set -eo pipefail
+
+# find a list of files whose names do not match our convention
+errant_files=$(find bin/ _test/ -iname '*\.sh' -exec basename {} \; | grep '[^a-z_.].*')
```

Comment:
I don't entirely understand the regex's intent here, but it looks like what it's actually checking is that there is a single character not in `[a-z_.]`, followed by any number of arbitrary other characters. That feels like an odd requirement to check. I think that if we're just detecting the presence of a forbidden character, we could simplify like

```suggestion
errant_files=$(find bin/ _test/ -iname '*\.sh' -exec basename {} \; | grep '[^a-z_.]')
```

Reply:
> That feels like an odd requirement to check.

Perfect correction to my regular expression naïveté!
I was trying to find any file name that has forbidden characters. So your fix makes it so!